src/effects/VST3/VST3Effect.h

```diff
@@ -0,0 +1,131 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file VST3Effect.h
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+
+#pragma once
+
+#include <pluginterfaces/gui/iplugview.h>
+#include <pluginterfaces/vst/ivstaudioprocessor.h>
+#include <public.sdk/source/vst/hosting/module.h>
+
+#include "EffectInterface.h"
+
+#include "SampleCount.h"
+#include "effects/Effect.h"
+#include "internal/ComponentHandler.h"
+
+namespace Steinberg
+{
+   namespace Vst
+   {
+      class IComponent;
+      class IEditController;
+      class IParameterChanges;
+      class IComponentHandler;
+   }
+}
+
+class ParameterChangesProvider;
+
+/**
+ * \brief Objects of this class connect Audacity with VST3 effects
+ */
+class VST3Effect final : public EffectUIClientInterface
+{
+   //Keep strong reference to a module while effect is alive
+   std::shared_ptr<VST3::Hosting::Module> mModule;
```

Comment:
I see shared_ptr, also many IPtr because you must use them with VST3 objects, and those are shared pointers.

But in fact only mModule, mEffectClassInfo, and mSetup are shared among copies of the object (which must be made for master-slave connections), which isn't obvious unless you read the copy constructor.

Make it more obvious here in the header with comments, and perhaps rearrange the sequence so the really shared members are separate from the really unique ones.